embedded-graphics/src/primitives/polyline/thick_points.rs

```diff
@@ -42,12 +52,21 @@ impl<'a> Iterator for ThickPoints<'a> {
     type Item = Point;
 
     fn next(&mut self) -> Option<Self::Item> {
-        if let Some(point) = self.points_iter.next() {
-            Some(point)
-        } else {
-            self.points_iter = self.triangle_iter.next().map(|t| t.mathematical_points())?;
-
-            self.next()
+        loop {
+            if let Some(point) = self.points_iter.next() {
+                if !ContainsPoint::contains(&self.prev_triangle, point)
+                    && !ContainsPoint::contains(&self.prev_triangle2, point)
+                    && !ContainsPoint::contains(&self.prev_triangle3, point)
```

Comment:
Technically speaking, one of this can be elided. When the joint turns right, prev_triangle2 check is redundant, and when it turns left, prev_triangle3 is. Also, probably both can be elided for mitered joints, but I haven't tried that.

prev_triangle is necessary though to avoid overdrawing the joining edges.

Reply:
Although I think it may be better to find a correct solution instead of hammering out the issues with the current naive one.